distribution/docs/src/main/resources/_contents/_developing/developing-intro-contents.adoc

```diff
@@ -0,0 +1,816 @@
+
+== ${branding} Development Guidelines
+
+[NOTE]
+====
+Development requires full knowledge of the ${branding} Catalog.
+====
+
+${branding} is written in Java and requires a moderate amount of experience with the Java programming language, along with Java terminology, such as packages, methods, classes, and interfaces.
+${branding} uses a small OSGi runtime to deploy components and applications.
+Before developing for ${branding}, it is necessary that developers have general knowledge on OSGi and the concepts used within.
+This includes, but is not limited to, Catalog Commands and the following topics:
+
+* The Service Registry
+** How services are registered
+** How to retrieve service references
+* Bundles
+** Their role in OSGi
+** How they are developed
+
+Documentation on OSGi can be viewed at the http://www.osgi.org[OSGi Alliance website].
+Helpful literature for beginners includes _OSGi and Apache Felix 3.0 Beginner's Guide_ by Walid Joseph Gédéon and _OSGi in Action: Creating Modular Applications in Java_ by Richard Hall, Karl Pauls, Stuart McCulloch, and David Savage.
+
+=== Recommended Hardware
+
+Because of its modular nature, ${branding} may require a few or many system resources, depending on which bundles and features are deployed.
+In general, ${branding} will take advantage of available memory and processors.
+A 64-bit JVM is required, and a typical installation is performed on a single machine with 16GB of memory and eight processor cores.
+
+The ${branding} source code is not tied to any particular IDE.
+However, if a developer is interested in setting up the Eclipse IDE, they can view the http://books.sonatype.com/m2eclipse-book/reference/[Sonatype guide] on developing with Eclipse.
+
+==== Getting Set Up
+
+To develop on ${branding}, access to the source code via Github is required.
+
+===== Integrated Development Environments (IDE)
+
+The ${branding} source code is not tied to any particular IDE.
+However, if a developer is interested in setting up the Eclipse IDE, they can view the http://books.sonatype.com/m2eclipse-book/reference/[Sonatype guide] on developing with Eclipse.
+
+==== Formatting Source Code
+
+A code formatter for the Eclipse IDE that can be used across all ${branding} projects will allow developers to format code similarly and minimize merge issues in the future.
```

Comment:
We should include a link to the ddf-support github project for the code formatting rules.

And here's where I expand on why we might want to push IJ over Eclipse...the Eclipse formatting config file had gotten incredibly stale so it's been deleted from the repo rather than be left around with out-of-date formatting rules. I _believe_ the xml file that can be extracted from the IJ formatting jar is Eclipse-compatible but I've never tested that.

We should test that and re-add an Eclipse formatting xml file to the ddf-support repo.